### DISCUSSION TITLE
Does source.on_metadata inherit the id?

### DISCUSSION THREAD

**Discussion by Loxbie** [savonet/liquidsoap#3872] [User Support]
Hey the title says it all: does `source.on_metadata()` inherit the id of the given source? In my example I have an input stream which I later on use in a mix. I also have server functions for this source, these are automatically created. But the namespace differs, some use the id I set on top `in_stream_0` and the last one has the id `on_metadata`.

```liq
input_stream_0 =
  input.http(
    id="in_stream_0",
    max_buffer=input_stream_max_buffer,
    timeout=input_stream_timeout,
    start=false,
    ""
  )

# Enable metadata insertion and store callbacks
input_stream_0 = insert_metadata(input_stream_0)
in_stream_insert_meta_0 = input_stream_0.insert_metadata
in_stream_last_meta_0 = input_stream_0.last_metadata

# Map metadata
def on_stream_meta_0(meta) =
  lm = (in_stream_last_meta_0() ?? [])
  merge_meta(lm, meta)
end
input_stream_0 = metadata.map(on_stream_meta_0, input_stream_0)

input_stream_0 = source.on_metadata(input_stream_0, on_metadata_notification)

print("#{source.id(input_stream_0)}")
```

The print statement gives me `on_metadata`, I expected `in_stream_0`. Is this intended?

**Comment by vitoyucepi** [comment]
Hi @Loxbie,
Yes, this behavior is intended, but you can set your own `id` parameter like this:
[`input_stream_0 = source.on_metadata(input_stream_0, on_metadata_notification, id="my_identifier")`](https://www.liquidsoap.info/doc-2.2.4/reference.html#source.on_metadata).
Alternatively, you can use the `on_metadata` method defined for source:
[`input_stream_0.on_metadata(on_metadata_notification)`](https://www.liquidsoap.info/doc-2.2.4/reference.html#input.http).

The [`on_metadata`](https://www.liquidsoap.info/doc-1.4.4/reference.html#on_metadata) function looks like a legacy function that existed before the records syntax was introduced.
